onnxruntime/core/providers/shared_library/provider_api.h

```diff
@@ -67,7 +80,16 @@ struct DeleteOnUnloadPtr {
 };
 
 constexpr const char* kOnnxDomain = "";
+constexpr const char* kMSDomain = "com.microsoft";
 constexpr const char* kDnnlExecutionProvider = "DnnlExecutionProvider";
+constexpr const char* kTensorrtExecutionProvider = "TensorrtExecutionProvider";
+
+enum CUDAStreamType : int {
+  kCudaStreamDefault = 0,
+  kCudaStreamCopyIn,
+  kCudaStreamCopyOut,
+  kTotalCudaStreams,
+};
```

Comment:
Are these copies of things defined in other places too? We need to obey ODR and avoid code duplication as well.

Reply:
Yeah, they're only defined once, but the providers can't include the main file that defines them as it pulls in incompatible dependencies. I'll put a comment.